documentation/CLI_usage.md

```diff
@@ -0,0 +1,195 @@
+# CLI Usage
+
+---
+## npd
+
+### Description
+napari plugin devtools scans the current python environment to find out details about napari plugin information.
+
+### Usage
+```
+npd [COMMANDS] [ARGS...]
+```
+
+### Commands
+|Command        |Description                                          |
+|---------------|-----------------------------------------------------|
+|validate       |validate the given package is correctly setup        |
+|discover       |discover all plugins under current python environment|
+
+
+## npd validate
+
+### Description
+Validate package(s) within current python environment, report PASSED or FAILED for each check, 
+only return 1 when all checks passed.
+
+### Usage
+```
+npd validate [Options] <package> [<packages>...]
+```
+
+### Options
+|name, shorthand    |Description                     |
+|-------------------|--------------------------------|
+|--verbose, -v      |Show more detailed output       |
+
+### Examples
+
+validate single package
+```
+$ pip install napari-svg
+$ npd validate napari-svg
+----------------------------------------------------------------
+Scanning current python environment: /Library/Frameworks/Python.framework/Versions/3.8
+----------------------------------------------------------------
+Validating package napari-svg
+* Classifier check: FAILED
+* Entrypoint check - syntax: PASSED
+* Entrypoint check - plugin registration: PASSED
+* Entrypoint check - registered modules exist: PASSED
+* Hooks registration check - svg: PASSED
+----------------------------------------------------------------
+Error: 'Framework :: napari' does not exist for napari-svg's 12 known classifier(s)
+----------------------------------------------------------------
+```
+
+validate multiple packages with verbose output:
+```
+$ pip install napari-svg
+$ pip install napari-demo
+$ npd validate napari-svg napari-demo -v
+----------------------------------------------------------------
+Scanning current python environment: /Library/Frameworks/Python.framework/Versions/3.8
+----------------------------------------------------------------
+Validating package napari-svg
+* Classifier check: FAILED
+* Entrypoint check - syntax: PASSED
+* Entrypoint check - plugin registration: PASSED
+        Plugins registerd under napari-svg: ['svg']
+* Entrypoint check - registered modules exist: PASSED
+        Modules found: ['napari_svg']
+* Hooks registration check - svg: PASSED
+        Found 6 registered writer hook(s) for svg:
```

Comment:
I like that we're listing out the hooks that were found now in verbose mode. This is helpful to see all the hooks that are registered.